### PR TITLE
Remove non-capture free substitution and add unit tests

### PR DESCRIPTION
**Pull request by nishantjr**

### CODE REVIEW

File: prover/lang/kore-lang.md [hunk 1]
```diff
@@ -461,60 +461,47 @@ and values, passed to K's substitute.
...
 ```k
   syntax PatternOrVarName ::= Pattern | VariableName
-
-  // capture-avoiding substitution
-  syntax Pattern ::= casubst(Pattern, PatternOrVarName, Pattern)    [function]
-
-  rule casubst(P, V, S) => #subst(true, P, V, S)
-
-  // non-capture-avoiding substitution
   syntax Pattern ::= subst(Pattern, PatternOrVarName, Pattern)    [function, klabel(subst)]
```
Comment (xc93) [inline prover/lang/kore-lang.md:474]: So `subst` only works for free variables, but it is not the capture-avoiding substitution. Is that right?

File: prover/lang/kore-lang.md [hunk 2]
```diff
@@ -552,12 +539,10 @@ Substitution: Substitute term or variable
   rule substPatternsMap(.Patterns, SUBST) => .Patterns
 
   // the Bool param: avoid capture
-  syntax Patterns ::= Patterns "[" Bool "," Pattern "/" Pattern "]" [function]
   syntax Patterns ::= Patterns "[" Pattern "/" Pattern "]" [function]
 
-  rule Ps[X/V] => Ps[false, X/V]
-  rule (BP, BPs)[CA, X/V] => #subst(CA,BP,X,V), (BPs[CA, X/V])
-  rule .Patterns[_, X/V] => .Patterns
+  rule (BP, BPs)[X/V] => subst(BP,X,V), (BPs[X/V])
```
Comment (xc93) [inline prover/lang/kore-lang.md]: This notation `P[X/V]` is different from what we usually use in the papers: `P[V/X]`, but it's OK. I suggest a comment here.

**Review by xc93 — APPROVED** [review]

File: prover/lang/kore-lang.md (hunk 1 above)
Comment (nishantjr) [reply prover/lang/kore-lang.md:474]: No, This commit removes the old `subst` and renames `casubst` to `subst`. We don't need non-capture avoiding substitution.

File: prover/lang/kore-lang.md (hunk 2 above)
Comment (nishantjr) [reply prover/lang/kore-lang.md]: I'd honestly like to get rid of the mix fix notation  for substitution completely. I find it really confusing.

**Pull request merged**